### PR TITLE
Avoid running loganalytics on CI tests

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/openshift-azure/issues/407

@openshift/sig-azure

### CODE REVIEW

**Comment by mjudeikis** [comment]
lgtm, the second pair of eyes for visibility

File: pkg/arm/arm.go [hunk 1]
```diff
@@ -43,16 +44,23 @@ func (*simpleGenerator) Generate(ctx context.Context, cs *acsapi.OpenShiftManage
...
+				return util.Template(string(masterStartup), startupMap, cs, map[string]interface{}{"Role": role})
 			} else {
-				return util.Template(string(nodeStartup), nil, cs, map[string]interface{}{"Role": role})
+				return util.Template(string(nodeStartup), startupMap, cs, map[string]interface{}{"Role": role})
 			}
 		},
 		"IsUpgrade": func() bool {
 			return isUpdate
 		},
+		"DevelopmentMode": devModeFunc,
```
Comment (jim-minter) [inline pkg/arm/arm.go]: put this straight in util.Template, then you don't need the startupMap?

File: pkg/arm/arm.go [hunk 2]
```diff
@@ -43,16 +44,23 @@ func (*simpleGenerator) Generate(ctx context.Context, cs *acsapi.OpenShiftManage
 	if err != nil {
 		return nil, err
 	}
+
+	devModeFunc := func() bool {
+		return os.Getenv("DEVELOPMENT_MODE") != ""
```
Comment (jim-minter) [inline pkg/arm/arm.go]: I'm on the fence about whether it makes sense to use the same env var for:
1- disabling LogAnalytics in at least most tests (permanent move; rationale: it's expensive)
2- enabling rotation for some dev/test? (temporary; rationale: getting more soak testing?)
I guess it's OK.

Clearly we want to minimise the number of these special cases over time.
Also would an env var like RUNNING_UNDER_TEST or something be a clearer name than DEVELOPMENT_MODE?

File: pkg/arm/data/master-startup.sh [hunk 3]
```diff
@@ -595,6 +595,7 @@ spec:
 EOF
 fi
 
+{{ if not DevelopmentMode }}
```
Comment (jim-minter) [inline pkg/arm/data/master-startup.sh]: `{{-` for all of these `if`s and `end`s

Comment (kargakis) [reply pkg/arm/data/master-startup.sh]: I've also seen this in azuredeploy.json. What's the difference?

Comment (pweil-) [reply pkg/arm/data/master-startup.sh]: eliminates whitespace

File: pkg/arm/arm.go (hunk 2 above)
Comment (mjudeikis) [reply pkg/arm/arm.go]: `DEVELOPMENT_MODE` is more clear for me than `RUNNING_UNDER_TEST` :) so question if perception :)

Comment (jim-minter) [reply pkg/arm/arm.go]: Yes, but it's not an environment variable that developers should really be using.

File: pkg/util/util.go [hunk 4]
```diff
@@ -29,6 +29,9 @@ func Template(tmpl string, f template.FuncMap, cs *acsapi.OpenShiftManagedCluste
 			replacer := strings.NewReplacer("$", "\\$")
 			return replacer.Replace(b)
 		},
+		"RunningUnderTest": func() bool {
+			return os.GetEnv("RUNNING_UNDER_TEST") != nil
```
[the remark this answers is not in the document]
Comment (kargakis) [reply pkg/util/util.go]: fixed, thanks

**Comment by jim-minter** [comment]
/lgtm

@kargakis please could you do a follow up which sets the env var in our testing?

**Comment by kargakis** [comment]
> @kargakis please could you do a follow up which sets the env var in our testing?

That's https://github.com/openshift/release/pull/1451

**Comment by jim-minter** [comment]
thanks! had missed it!

**Pull request merged**